src/App.tsx

```diff
@@ -1,24 +1,234 @@
+import { TodoList } from './components/TodoList/TodoList';
+import { Status } from './types/Status';
+import { Errors } from './types/Errors';
+import { TodosFilter } from './components/TodoFilter/TodoFilter';
 
-const USER_ID = 0;
+const USER_ID = 7023;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredType, setFilteredType] = useState(Status.All);
+  const [, setTodo] = useState<Todo | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [title, setTitle] = useState<string>('');
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleToggleAll = () => {
+    const allCompleted = todos.every(todo => todo.completed === true);
+
+    if (allCompleted) {
+      const updatedTodos = todos.map(todo => ({
+        ...todo,
+        completed: false,
+      }));
+
+      setTodos(updatedTodos);
+    } else {
+      const updatedTodos = todos.map(todo => ({
+        ...todo,
+        completed: true,
+      }));
+
+      setTodos(updatedTodos);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setErrorMessage(Errors.EMPTY_TITLE);
+    }
+
+    setIsInputDisabled(true);
+    const newTodo: Todo = {
+      title: title.trim(),
+      userId: USER_ID,
+      id: +Date,
+      completed: false,
+    };
+
+    setTodo(newTodo);
+
+    addTodo({ title: title.trim(), userId: USER_ID, completed: false })
+      .then(todo => setTodos((currentTodos) => [...currentTodos, todo]))
+      .catch(() => {
+        setTitle(title);
+        setErrorMessage(Errors.CAN_NOT_ADD_TODO);
+      })
+      .finally(() => {
+        setTodo(null);
+        setIsInputDisabled(false);
+      });
+
+    setTitle('');
+  };
+
+  const handleToggleTodo = (id: number) => {
+    const updatedTodos = todos.map(currentTodo => {
+      if (currentTodo.id === id) {
+        return { ...currentTodo, completed: !currentTodo.completed };
+      }
+
+      return currentTodo;
+    });
+
+    setTodos(updatedTodos);
+  };
+
+  const handleDeleteTodo = (todo: Todo) => {
+    deleteTodo(todo.id)
+      .catch(() => setErrorMessage(Errors.CAN_NOT_DELETE_TODO))
+      .finally(() => {
+        setTodos(current => current.filter(item => item.id !== todo.id));
+      });
+  };
+
+  const handleDeleteCompleted = async () => {
+    const deletedTodo = todos.filter(todo => todo.completed);
+    const needToDelete = deletedTodo.map(todo => {
+      return deleteTodo(todo.id);
+    });
+
+    try {
+      await Promise.allSettled(needToDelete);
+      const newTodos = todos.filter(todo => !todo.completed);
+
+      setTodos(newTodos);
+    } catch {
+      () => {
+        setErrorMessage(Errors.CAN_NOT_DELETE_TODO);
+      };
+    }
+  };
+
+  const filterTodos = useMemo(() => {
```

Comment:
```suggestion
  const filteredTodos = useMemo(() => {
```
`filterTodos` with verb in the name sounds like a function name